nvme.c

```diff
@@ -2478,22 +2478,35 @@ static int create_ns(int argc, char **argv, struct command *cmd, struct plugin *
 	return err;
 }
 
-static bool nvme_match_device_filter(nvme_subsystem_t s)
+static bool nvme_match_device_filter(nvme_subsystem_t s,
+		nvme_ctrl_t c, nvme_ns_t ns, void *f_args)
 {
-	nvme_ctrl_t c;
-	nvme_ns_t n;
+	int ret, instance, nsid, s_num;
+	char *devname = f_args;
 
-	if (!devicename || !strlen(devicename))
+	if (!devname || !strlen(devname))
 		return true;
 
-	nvme_subsystem_for_each_ctrl(s, c) {
-		if (!strncmp(devicename, nvme_ctrl_get_name(c),
-			     strlen(nvme_ctrl_get_name(c))))
+	ret = sscanf(devname, "nvme%dn%d", &instance, &nsid);
+	if (ret != 2)
+		return true;
+
+	if (s) {
+		ret = sscanf(nvme_subsystem_get_name(s), "nvme-subsys%d",
```

Comment:
So we don't have any way to follow any links in sysfs and instead we have to match numbers in the names?